src/test/ui/borrowck/borrowck-union-borrow.nll.stderr

```diff
@@ -1,132 +1,121 @@
 error[E0502]: cannot borrow `u.a` as mutable because it is also borrowed as immutable
-  --> $DIR/borrowck-union-borrow.rs:27:23
+  --> $DIR/borrowck-union-borrow.rs:25:23
    |
 LL |             let ra = &u.a;
    |                      ---- immutable borrow occurs here
-LL |             let rma = &mut u.a; //[ast]~ ERROR cannot borrow `u.a` as mutable because it is also borrowed as immutable
+LL |             let rma = &mut u.a; //~ ERROR cannot borrow `u.a` as mutable because it is also borrowed as immutable
    |                       ^^^^^^^^ mutable borrow occurs here
-LL |                                 //[mir]~^ ERROR cannot borrow `u.a` as mutable because it is also borrowed as immutable
 LL |             drop(ra);
    |                  -- immutable borrow later used here
 
 error[E0506]: cannot assign to `u.a` because it is borrowed
-  --> $DIR/borrowck-union-borrow.rs:33:13
+  --> $DIR/borrowck-union-borrow.rs:30:13
    |
 LL |             let ra = &u.a;
    |                      ---- borrow of `u.a` occurs here
-LL |             u.a = 1; //[ast]~ ERROR cannot assign to `u.a` because it is borrowed
+LL |             u.a = 1; //~ ERROR cannot assign to `u.a` because it is borrowed
    |             ^^^^^^^ assignment to borrowed `u.a` occurs here
-LL |                      //[mir]~^ ERROR cannot assign to `u.a` because it is borrowed
 LL |             drop(ra);
    |                  -- borrow later used here
 
-error[E0502]: cannot borrow `u.b` as mutable because it is also borrowed as immutable
-  --> $DIR/borrowck-union-borrow.rs:50:23
+error[E0502]: cannot borrow `u` (via `u.b`) as mutable because it is also borrowed as immutable (via `u.a`)
```

Comment:
"via" seems like an improvement, but I wonder if we can do better. We're not -- for example -- telling the user that `a` and `b` are both union fields.

I wonder if we could just add a *note* or something? Like, keep the original message, but say:

```
= note: `u.a` is a field of the union `Foo`, so it overlaps the field `u.b`
```

or maybe also edit the messages talking about the intersecting path?

```
error[E0506]: cannot assign to `u.b` because it is borrowed
  --> $DIR/borrowck-union-borrow.rs:30:13
   |
LL |             let ra = &u.a;
   |                      ---- borrow of `u.a` -- which overlaps `u.b` -- occurs here
LL |             u.b = 1; //~ ERROR cannot assign to `u.a` because it is borrowed
   |             ^^^^^^^ assignment to borrowed `u.b` occurs here
LL |             drop(ra);
   |                  -- borrow later used here
   |
   = note: `u.a` is a field of the union `Foo`, so it overlaps the field `u.b`
```

Reply:
Pushed a commit that makes this change.